samples_skills/conversationalRetrievalwithLangchain.ipynb

```diff
@@ -0,0 +1,266 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we will demonstrate how to use LangChain to perform [conversational retrieval](https://python.langchain.com/docs/use_cases/question_answering/chat_vector_db).\n",
+    "\n",
+    "In a conversational question and answering scenario, users often pose follow-up questions related to the same topic, with the context being crucial to understand their queries. To address such cases effectively, we use the ConversationalRetrievalChain. Behind the scenes, this chain takes the user's question and converts it into a standalone query by considering the conversation history. Subsequently, it uses this standalone question to query the search service for relevant information."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install azure-identity"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook we are utilizing an existing search index we already set up with Azure Cognitive Search. You can follow this [link](https://python.langchain.com/docs/integrations/vectorstores/azuresearch) to create your own search index. In our search index, the content vector field is named \"contentVector\", so we set it as an environment variable below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"AZURESEARCH_FIELDS_CONTENT_VECTOR\"] = \"contentVector\""
```

Comment:
rag_prop_j_3 does not use the name contentVector, it uses `Embedding`

https://ms.portal.azure.com/#view/Microsoft_Azure_Search/Index.ReactView/id/%2Fsubscriptions%2Fc1f2cd4b-4e91-4a9f-9211-07df640f8610%2FresourceGroups%2FAppliedAI%2Fproviders%2FMicrosoft.Search%2FsearchServices%2Fbrchuncognitivesearch%23rag_prop_j_3/location/West%20US%202/sku/standard

Reply:
Ah I see you aren't using the `COGSEARCH_INDEX_NAME` in the `llm.env`, you are using the code snippet with index_name = "testqa".

Reply:
Yeah, I think we shouldn't have the index name in the llm env as it could change depending on the sample. I defined mine in the notebook as you noticed.